### ISSUE TITLE
tvOS: timeshift does not work

### ISSUE THREAD

**Issue by thenktor** [zipleen/tvheadend-iphone-client#216]
Hi,

pressing the play/pause on the Apple TV remote IMHO should pause playback but it does nothing.

**Comment by zipleen** [comment]
I'm sorry if you though time shift is available, although I'm really curious to know where you got the impression that time shift _should_ work. 

The issue is that time shift option is not available in HTTP streaming via tvheadend - only through HTSP - and vlc only support HTTP streaming.

If any of these two things change, then tvhclient will be able to support time shift.

**Comment by thenktor** [comment]
Hi,

I could not find anything about it in the known problems section in the app so I did open this bug report / feature request.
IMHO it's one of the most important features :(

**Comment by zipleen** [comment]
I will add specific information that timeshift is not available to that screen, and also to the in app purchase one.

I agree that timeshift is important, however there's very little I can do on the time that I have available. Any help would be very much appreciated though!

**Comment by tonyschuite** [comment]
apparently it could theoretically work:
https://github.com/BtbN/vlc-htsp-plugin

**Comment by zipleen** [comment]
@tonyschuite check this conversation from 21 march 2017 - https://gitter.im/zipleen/tvheadend-iphone-client?at=58d1199ffe6a638b1aea70d6

**Comment by mpietruschka** [comment]
https://trac.videolan.org/vlc/ticket/19553

Doesn't  "--http-continuous" mean local timeshift?

**Comment by mpietruschka** [comment]
VLC HTS Plugin (Legacy)
https://github.com/BtbN/vlc-htsp-plugin

**Comment by zipleen** [comment]
yup, I never said that VLC does not support timeshift, I said that tvheadend must support timeshift through the HTTP protocol =)

In relation to the htsp plugin for vlc, it's very old code and I would need to hack VLCKit in order to built it in. One thing that I learned is that I do _not_ want to ever again have to build VLCKit at all. It's either supported officially bundled with VLCKit, or nothing. 

(However there's a way to use HTSP and get vlc to load bytes of stream data instead of giving it a http url :P)

**Comment by mpietruschka** [comment]
Couln't you just enable the local timeshift in VLC and enable the pause button in the UI?

**Comment by zipleen** [comment]
there's no local timeshift in vlc :D 

at least, from what I know. the only thing that exists is the in memory cache of vlc, which is basically a few seconds depending on the option you select for how big you want the network cache to be.

if there's any sort of local timeshift in vlc, do let me know how to enable it ! 

but first, open VLC for iOS and attempt to see if they have it in any way (a few seconds doesn't count, that's just the in memory cache - SD content could have up to 10sec of cache if you have a big enough network cache)

**Comment by zipleen** [comment]
I've got a proposal. What about opening a bounty for adding htsp protocol with timeshift support in https://www.bountysource.com ? 

There's basically multiple ways to achieve this:
- add HTSP with timeshift support to VLCKit (we can open a ticket for that one and attach the bounty to it - this can be done by anyone)
- add HTSP support in tvhclient, so it uses the binary VLCKit protocol to stream data (this one can only be done by me)
- create an open source iOS project that uses VLCKit (with a compatible license like BSD) to watch htsp streams supporting timeshift (this one can be done by anyone, I would then just use that source in tvhclient)

The fact is that I don't use timeshift at all in tvheadend, so I have very little incentive to add this feature. If you have a bounty, we can see how many people would actually pay to get this feature! 

(not being pessimistic, but I would bet a very very small amount of people would care hehe)

**Comment by Nastras** [comment]
Hello @zipleen,

i would also be very happy about the timeshift function for the Apple TV and would be willing to pay for it again .-)

So I would like to know what the current status is?

**Comment by zipleen** [comment]
Hey!

I haven't done anything for this, there's way more things to do if I have time : dark theme support, fix the recordings screen which I'm in the middle of and it would be nice to maybe look at that EPG in the Apple TV version.

Timeshift is way down in the priority list..

**Comment by Nastras** [comment]
Thanks for the quick answer. 

I've only been using your apps for Apple TV and iOS since today and I'm thrilled. Very good work!

That sounds good and bad. I'm glad you're still developing👍
If you ever get bored, Timeshift would be a cool extension to make your app complete 😉

Greetings Nastras
